interfaces/cython/cantera/composite.py

```diff
@@ -516,32 +516,38 @@ def __init__(self, phase, shape=(0,), states=None, extra=None):
         reserved = self.__dir__()
 
         self._extra = {}
-        if isinstance(extra, dict):
-            for name, v in extra.items():
+
+        if isinstance(extra, str):
+            extra = [extra]
+
+        if isinstance(extra, (list, tuple, np.ndarray)) and all(isinstance(name, str) for name in extra):
+            #if extra is a list and all elements in extra are of type str
+            for name in extra:
                 if name in reserved:
                     raise ValueError(
                         "Unable to create extra column '{}': name is already "
                         "used by SolutionArray objects.".format(name))
-                if not np.shape(v):
-                    self._extra[name] = [v]*self._shape[0]
-                elif len(v) == self._shape[0]:
-                    self._extra[name] = list(v)
-                else:
-                    raise ValueError("Unable to map extra SolutionArray"
-                                     "input for named {!r}".format(name))
+                self._extra[name] = np.empty(self._shape)
 
-        elif extra and self._shape == (0,):
-            for name in extra:
+        elif isinstance(extra, dict):
+            for name, v in extra.items():
                 if name in reserved:
                     raise ValueError(
                         "Unable to create extra column '{}': name is already "
                         "used by SolutionArray objects.".format(name))
-                self._extra[name] = []
+                if not np.shape(v):
+                    self._extra[name] = np.array([v]*self._shape)
+                elif len(v) == self._shape or np.array(v).shape == self._shape:
+                    self._extra[name] = np.array(list(v))
```

Comment:
I don't think you need the `list(v)` here, since you're converting it to an `np.array()` in the check, `v` should already be able to be directly converted. Plus, converting to a list may change the dimensions.

Reply:
yes, this was relevant more to the original issue #829 .I guess not anymore as `extra` beyond that.

will remove this.